File: exercism/python/raindrops/raindrops.py
"""no module docstring"""

import sys

def convert(number):
    """convert -> return raindrop sounds based on input"""

    a_dict = {3:'Pling', 5:'Plang', 7:'Plong'}
    # https://stackoverflow.com/questions/9060653/list-comprehension-without-in-python
    ret = ''.join((v for k, v in a_dict.items() if int(number) % k == 0))
    return ret if ret else str(number)

if __name__ == "__main__":
    print(convert(sys.argv[1]))

File: exercism/kotlin/raindrops/src/main/kotlin/Raindrops.kt
object Raindrops {

    // returns raindrop sounds
    fun convert(n: Int): String {
        val sounds = mapOf(3 to "Pling", 5 to "Plang", 7 to "Plong")
            .filter { n % it.key == 0 }
            .map{ it.value }
            .joinToString("")
        return if(sounds.isEmpty()) "${n}" else sounds
    }
}
